File: IPaXMLSection/IPaHTMLSection.m
//
//  IPaHTMLSection.m
//  IPaXMLSection
//
//  Created by IPaPa on 13/5/14.
//
//

#import "IPaHTMLSection.h"
#include <libxml/HTMLparser.h>
@implementation IPaHTMLSection
-(id)initWithXMLData:(NSData*)data
{
    htmlDocPtr doc = htmlParseDoc((xmlChar*)[data bytes], "UTF-8");
    if (doc == nil) {
        NSLog(@"Failed to parse html");
        return nil;
    }
    xmlNodePtr root = xmlDocGetRootElement(doc);
    
    return [super initWithXMLNode:root];
}
-(id)initWithXMLFile:(NSString*)fileName
{
    htmlDocPtr doc = htmlParseFile([fileName UTF8String], "UTF-8");
    if (doc == nil) {
        NSLog(@"Failed to parse html");
        return nil;
    }
    xmlNodePtr root = xmlDocGetRootElement(doc);
    
    return [super initWithXMLNode:root];
    
}
@end

File: Sources/IPaXMLSection/IPaHTMLSection.swift
//
//  IPaHTMLSection.swift
//  IPaXMLSection
//
//  Created by IPa Chen on 2020/7/21.
//

import UIKit
import libxml2
import IPaLog
open class IPaHTMLSection: IPaXMLSection {
    public override init?(_ xmlData: Data) {
        
        let nsXmlData = xmlData as NSData
        guard let utf8Data = "UTF-8".data(using: .utf8, allowLossyConversion: false) ,let doc = htmlParseDoc(nsXmlData.bytes.bindMemory(to: xmlChar.self, capacity: 1), (utf8Data as NSData).bytes.bindMemory(to: Int8.self, capacity:1)),let root = xmlDocGetRootElement(doc) else {
            IPaLog("Failed to parse html!")
            return nil
        }
        super.init(root)
    }
    public override init?(_ fileName: String) {
        guard let utf8Data = "UTF-8".data(using: .utf8, allowLossyConversion: false), let doc = htmlParseFile(fileName.cString(using: .utf8), (utf8Data as NSData).bytes.bindMemory(to: Int8.self, capacity:1)),let root = xmlDocGetRootElement(doc) else {
            IPaLog("Failed to parse html!")
            return nil
        }
        super.init(root)
    }
    
}
